src/table/components/head/HeadCellMenu.tsx

```diff
@@ -1,110 +1,145 @@
+              setOpenPrimaryDropdown(false);
+            },
+            icon: <ArrowUpwardIcon />,
+            isDisabled: !isInteractionEnabled || (isCurrentColumnActive && sortDirection === 'A'),
+          },
+          {
+            id: 2,
+            itemTitle: translator.get('SNTable.MenuItem.SortDescending'),
+            onClick: (evt: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+              sortFromMenu(evt, 'D');
+              setOpenPrimaryDropdown(false);
+            },
+            icon: <ArrowDownwardIcon />,
+            isDisabled: !isInteractionEnabled || (isCurrentColumnActive && sortDirection === 'D'),
+          },
+        ],
+      },
+      ...(isDimension
+        ? [
+            {
+              id: 2,
+              options: [
+                {
+                  id: 1,
+                  itemTitle: translator.get('SNTable.MenuItem.Search'),
+                  onClick: (evt: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+                    evt.stopPropagation();
+                    setOpenPrimaryDropdown(false);
+                    setOpenSecondaryDropdown(true);
+                  },
+                  icon: <SearchIcon />,
+                  isDisabled: false,
+                },
+              ],
+            },
+          ]
+        : []),
+    ],
+    [translator, isInteractionEnabled, isCurrentColumnActive, sortDirection, isDimension]
+  );
 
   return (
     <StyledCellMenu headerStyle={headerStyle}>
       <StyledMenuIconButton
         ref={anchorRef}
         tabIndex={-1}
         id="sn-table-head-menu-button"
-        aria-controls={open ? 'sn-table-head-menu' : undefined}
-        aria-expanded={open ? 'true' : undefined}
+        aria-controls={openPrimaryDropdown ? 'sn-table-head-menu' : undefined}
+        aria-expanded={openPrimaryDropdown ? 'true' : undefined}
         aria-haspopup="true"
-        onClick={() => setOpen(!open)}
+        onClick={() => setOpenPrimaryDropdown(!openPrimaryDropdown)}
       >
         <MoreHoriz />
       </StyledMenuIconButton>
       <Popper
-        modifiers={[
-          {
-            name: 'offset',
-            options: {
-              offset: [0, 9],
-            },
-          },
-        ]}
-        open={open}
+        modifiers={[{ name: 'offset', options: { offset: [0, 9] } }]}
+        open={openPrimaryDropdown}
         anchorEl={anchorRef.current}
         role={undefined}
         placement="bottom-start"
         transition
         disablePortal
       >
-        {({ TransitionProps, placement }) => (
-          <Grow
-            {...TransitionProps}
-            style={{
-              transformOrigin: placement === 'bottom-start' ? 'left top' : 'left bottom',
-            }}
-          >
-            <Paper sx={{ boxShadow: 15 }}>
-              <ClickAwayListener onClickAway={() => setOpen(false)}>
+        {({ TransitionProps }) => (
+          <Grow {...TransitionProps} style={{ transformOrigin: 'left top' }}>
```

Comment:
Because there was no need for it at the first place, since the placement is explicitly been set to `        placement="bottom-start"` in a few lines above in `<Popper />` component and that placement is always going to give us the value of "bottom-start" (as I tested several times in different viewport sizes and different datasets). 

So the value is not going to change and that check is going to be unnecessary.

Reply:
ok great :)